### PR TITLE
HSD8-767 D7 News importer

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Migration for D7 News content

# Need Review By (Date)
- ASAP

# Urgency
- medium

# Steps to Test
1. Checkout this branch
1. either install a new site or run `blt drupal:update --site=[sitename]`
1. navigate to `/admin/structure/migrate/news`
1. enter `https://internationalrelations.stanford.edu/stanford-news/news-xml` as the uurl 
1. run cron
1. validate news content was created.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
@sherakama is it possible to review this today please? H&S said it's a blocker for a site at this time.

File: docroot/modules/humsci/hs_migrate/src/Overrides/MigrationOverrides.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+   */
+  public function getCacheableMetadata($name) {
+    return new CacheableMetadata();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheSuffix() {
+    return 'CourseImporterOverrides';
```
Comment (sherakama) [inline docroot/modules/humsci/hs_migrate/src/Overrides/MigrationOverrides.php]: Courses? Should this be news?

Comment (pookmish) [reply docroot/modules/humsci/hs_migrate/src/Overrides/MigrationOverrides.php]: copy pasta strikes again

File: docroot/modules/humsci/hs_migrate/src/Overrides/MigrationOverrides.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+   */
+  public function createConfigObject($name, $collection = StorageInterface::DEFAULT_COLLECTION) {
+    return NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheableMetadata($name) {
+    return new CacheableMetadata();
```
Comment (sherakama) [inline docroot/modules/humsci/hs_migrate/src/Overrides/MigrationOverrides.php:68]: Is this good enough to be empty?

File: docroot/modules/humsci/hs_migrate/hs_migrate.module [hunk 3]
```diff
@@ -31,6 +31,19 @@ function hs_migrate_help($route_name, RouteMatchInterface $route_match) {
...
+ *
+ * @param mixed $arg
+ *   Argument passed from the callback migrate process plugin.
+ *
+ * @return int
+ *   Current time.
+ */
+function _hs_migrate_get_time($arg = NULL) {
+  return time();
+}
```
Comment (sherakama) [inline docroot/modules/humsci/hs_migrate/hs_migrate.module:52]: Why is this needed?

Comment (pookmish) [reply docroot/modules/humsci/hs_migrate/hs_migrate.module:52]: i'll document like i did in the cardinal service profile
see https://github.com/SU-HKKU/cardinal_service_profile/blob/8.x-1.x/cardinal_service_profile.profile#L36

> We use this function in migration callback processes because using `time` as the callback produces messages about "function accepts 0 arguments, 1 argument passed". So we just have our own callback that takes the argument from the migration process and does nothing with it.

File: docroot/modules/humsci/hs_migrate/src/Overrides/MigrationOverrides.php (hunk 2 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_migrate/src/Overrides/MigrationOverrides.php:68]: I haven't found where this is used yet

File: docroot/modules/humsci/hs_migrate/hs_migrate.module (hunk 3 above)
Comment (sherakama) [reply docroot/modules/humsci/hs_migrate/hs_migrate.module:52]: Thanks.

**Review by sherakama — APPROVED** [review]

**Comment by sherakama** [comment]
Approved. Please resolve conflicts and let the tests pass.
